practices/package_management.md

```diff
@@ -0,0 +1,82 @@
+
+Here are some examples of package managers:
+- [apk-tools](http://apk-tools.sourceforge.net/) for UNIX like RAM-based embedded systems
+- [dpkg](https://wiki.debian.org/Teams/Dpkg) for Debian (and now Ubuntu)
+- [pacman]() for DeLi Linux
+- [Steam](https://store.steampowered.com/) - a game platfor, works on Windows NT, OS X, and Linux
+- [Mac App Store](https://www.apple.com/) - official digital distribution platform for OS X apps
+- [NuGet](https://www.nuget.org/) - for Windows, available as a plugin for Visual Studio
+
+## Why You Might Want the Package Management
+
+Package managers
+- simplify work for developers
+- allow you to upgrade the packages very fast, you do not have to upgrade one by one
+- save the time
+- also have some advanced features - for example, they can make a basic audit
+- can maintain the project versions (automatically upload them online) so everyone uses the newest version
+- allow you to manage someone else’s open-source code with certainty that the build is consistent within the developers and environments
+
+## Issues the Package Management Solves
+
+- [Increased cost](/issues/increased-cost)
+- [Poor code quality](/issues/poor-code-quality)
+- [Meaningless work](/issues/meaningless-work)
+
+## How to Implement the Package Management
+
+1. Usually, the user requests a package using the package manager (PM).
+2. The PM finds and downloads the required package from a software repository. Those repositories are available online and every PM has associated configuration files that point to repository location.
+3. The PM installs the package and advises necessary manual steps.
+
+## Common Pitfalls of the Package Management
+
+- The package is not specified and when the new version is published, the package breaks. You have to specify the major version at least.
+- The dependencies are not updated. Important security bugs fixes could be missing.
+- The packages are downloaded and then their code is modified by hand. The changes disappear after the next installation.
+- The dependency security audit is being ignored.
+- Many different packages are being used or each developer uses a different package. They can contain different properties and act differently.
+- Libraries needed for development are added to the global dependencies.
+- Lock File is not respected.
+
+**What Is a File Locking**
+
+File locking is a useful feature for restricting other users from changing a specific computer file. The locked file can be managed only by one person at a time. This prevents interceding updates on the same files. Modern file locking systems allow multiple users to access the file but only the first one can modify it. Not respecting the lock file means that if the developer makes some changes and a previous file version is locked, the changes are not saved and the other users are going to download a not updated version (the locked one).
```

Comment:
This block makes no sense. This "file locking" described in the block is not the same as lockfile for package manager.

There are some articles about lockfiles in JavaScript package managers:
- https://www.vzhurudolu.cz/prirucka/package-lock-json
- https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/
- https://docs.npmjs.com/files/package-lock.json

Reply:
I see, thank you for the clarification. I have changed the paragraph, please let me know if it's ok.